syntax = "proto3";

service Sysinfo { 
    rpc GetSysInfo(SysInfo) returns (Response){}
    rpc GetProcesses(stream Process) returns (Response){}
}

message Response {
    string ip = 1;
    string action = 2;
}

message Process{
    string ip = 1;
    int32 pid = 2;
    string name = 3;
    float cpu = 4;
    float mem = 5;
}

message SysInfo{
    string ip = 1;
    float cpu = 2;
}
